components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -568,28 +568,51 @@ class GeckoEngineSessionTest {
     }
 
     @Test
-    fun `keeps track of current url via onPageStart events`() {
+    fun `do not keep track of current url via onPageStart events`() {
         val engineSession = GeckoEngineSession(mock(),
                 geckoSessionProvider = geckoSessionProvider)
 
         captureDelegates()
 
         assertNull(engineSession.currentUrl)
         progressDelegate.value.onPageStart(geckoSession, "https://www.mozilla.org")
-        assertEquals("https://www.mozilla.org", engineSession.currentUrl)
+        assertNull(engineSession.currentUrl)
 
         progressDelegate.value.onPageStart(geckoSession, "https://www.firefox.com")
+        assertNull(engineSession.currentUrl)
+    }
+
+    @Test
+    fun `keeps track of current url via onPageStart events`() {
```

Comment:
flyby: nit: this should be `onLocationChange` :)

Reply:
oops, thanks for catching this. 👍  I'll fix